### DISCUSSION TITLE
columns not working

### DISCUSSION THREAD

**Discussion by Manubi** [tailwindlabs/tailwindcss#8930] [Help] [answered]
Hi,

I've a question why the `columns-3` class is not keeping the book together with the title and author? Imho this shouldn't happen. Any ideas are appreciated. Thanks!
```
      <div className="gap-8 mt-4 columns-3">
        {books.results.map(book => (
          <div key={book.id} className="relative my-10 group">
            <div className="overflow-hidden rounded-md group-hover:opacity-75">
              <img
                src={book.cover_image_url}
                alt={book.title}
                className="rounded-md "
              />
            </div>
            <h3 className="mt-3 text-sm text-gray-500">
              <span className="" />
              {book.author}
            </h3>
            <p className="text-lg font-medium text-gray-900 dark:text-white">
              {book.title}
            </p>
          </div>
        ))}
      </div>
```
![CleanShot 2022-07-22 at 12 53 18](https://user-images.githubusercontent.com/20935138/180424802-f0531dd2-b576-4b39-ae0d-650af0470b22.png)

**Comment by Manubi** [comment]
Another problem with columns... why is it not trying to fill the space evenly?
![CleanShot 2022-07-22 at 16 45 53](https://user-images.githubusercontent.com/20935138/180464402-6c75fd46-d57a-4035-b6bc-5781d2ed0701.png)

**Comment by ichenglin** [comment]


**Reply by Manubi** [reply]
I am trying to get a masonry component such as pinterest. As mentioned in the code above I don't use grid but, `columns-3`.  Thanks for the help!
eg. https://paulcollett.github.io/react-masonry-css/demo/

**Comment by wongjn** [comment, marked as answer]
To keep the book and the author together, use [`break-inside-avoid`](https://tailwindcss.com/docs/break-inside) on each of the top-level `<div>` elements in the `.map()`. This is uses the [`break-inside` CSS property](https://developer.mozilla.org/docs/Web/CSS/break-inside) and for `avoid`, it "Avoids any break (page, column, or region) from being inserted within the principal box."
